extensions/src/AWSSDK.Extensions.CborProtocol/Internal/CborStreamReader.cs

```diff
@@ -247,26 +239,42 @@ public CborReaderState PeekState()
         {
             return ExecuteRead(r =>
             {
-                try
+                // We need to Peek twice incase the first time failed because we are near the end of the current chunk and we just need to refill.
+                for (int attempt = 0; attempt < 2; attempt++)
                 {
-                    return r.PeekState();
+                    try
+                    {
+                        var state = r.PeekState();
+                        if (state == CborReaderState.Finished && _nestingStack.Count > 0)
+                        {
+                            _logger.DebugFormat("PeekState returned Finished, but nesting stack is not empty. Attempting refill.");
+                            RefillBuffer(0);
+                            continue;
+                        }
+
+                        return state;
+                    }
+                    catch (CborContentException ex)
```

Comment:
Confirming you don't need to also catch the `InvalidOperationException` exception like you did above?